components/server/src/workspace/headless-log-service.ts

```diff
@@ -145,14 +145,20 @@ export class HeadlessLogService {
         logCtx: LogContext,
         instanceId: string,
         logEndpoint: HeadlessLogEndpoint,
+        cancel?: (retry: boolean) => void,
     ): Promise<HeadlessLogUrls | undefined> {
-        const tasks = await this.supervisorListTasks(logCtx, logEndpoint);
+        const tasks = await this.supervisorListTasks(logCtx, logEndpoint, cancel);
         return this.renderTasksHeadlessLogUrls(logCtx, instanceId, tasks);
     }
 
-    protected async supervisorListTasks(logCtx: LogContext, logEndpoint: HeadlessLogEndpoint): Promise<TaskStatus[]> {
+    protected async supervisorListTasks(
+        logCtx: LogContext,
+        logEndpoint: HeadlessLogEndpoint,
+        cancel?: (retry: boolean) => void,
+    ): Promise<TaskStatus[]> {
         if (logEndpoint.url === "") {
             // if ideUrl is not yet set we're too early and we deem the workspace not ready yet: retry later!
+            cancel?.(false);
```

Comment:
@geropl the comment above indicates we should retry in this case? Or are we handling that separately?